app/models/user.rb

```diff
@@ -12,27 +12,26 @@ class User < ApplicationRecord
   has_many :friendships
   has_many :inverse_friendships, class_name: :Friendship, foreign_key: :friend_id
 
-  def friends
-    friends_array = friendships.map { |friendship| friendship.friend if friendship.status == true }
-    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.status }
-    friends_array.compact
-  end
-
-  def pending_friends
-    friendships.map { |friendship| friendship.friend unless friendship.status }.compact
-  end
-
-  def friend_requests
-    inverse_friendships.map { |friendship| friendship.user unless friendship.status }.compact
-  end
+  has_many :pending_friendships, -> { where status: false }, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :pending_friends, through: :pending_friendships, source: :friend
+  has_many :confirmed_friend, -> { where status: true }, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :friends, through: :confirmed_friend, source: :friend
+  has_many :inverted_friendships, -> { where status: false }, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friend_requests, through: :inverted_friendships, source: :user
 
   def confirm_friend(user)
-    friendship = inverse_friendships.find { |friend| friend.user == user }
+    friendship = inverse_friendships.where(user_id: user.id).first
     friendship.status = true
     friendship.save
+    reverse_confirmation = friendships.build(friend_id: user.id, status: true)
+    reverse_confirmation.save
   end
```

Comment:
* [ ] You have the same method (better) in Friendship model, kindly delete this one.

Reply:
Please where in the friendship model are you referring to (confirmed_friend)?